src/test/scala/com/touk/hiring_task/multiplex/service/CinemaServiceSpec.scala

```diff
@@ -0,0 +1,270 @@
+package com.touk.hiring_task.multiplex.service
+
+import com.touk.hiring_task.multiplex.CinemaConstants
+import com.touk.hiring_task.multiplex.commons._
+import com.touk.hiring_task.multiplex.dto.{DetailsDto, MovieDto, PaymentDto, PlaceDto}
+import com.touk.hiring_task.multiplex.model.{Key, Place, Projection, Reservation}
+import com.touk.hiring_task.multiplex.repositories.{CinemaRepositoryImpl, FillCinemaRoomsServiceImpl}
+import com.touk.hiring_task.multiplex.utils.ValidatorImpl
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.concurrent.ScalaFutures.convertScalaFuture
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CinemaServiceSpec extends AnyFreeSpec with Matchers with MockFactory {
```

Comment:
Do you have an idea how to write an end-to-end test? To be sure that our app can be run and test all our use cases from the beginning to the end automatically?